packages/squiggle-lang/__tests__/Distributions/Invariants/Means_test.res

```diff
@@ -53,88 +30,115 @@ describe("Mean", () => {
   }
-  let combinations = E.L.combinations2(distributions)
-  let zipDistsDists = E.L.zip(distributions, distributions)
+  let pairsOfDifferentDistributions = E.L.combinations2(distributions)
+
+  let runMean: DistributionTypes.genericDist => float = dist => {
+    dist->mean->run->toFloat->E.O2.toExn("Shouldn't see this because we trust testcase input")
+  }
 
   let testOperationMean = (
-      distOp: (DistributionTypes.genericDist, DistributionTypes.genericDist) => result<DistributionTypes.genericDist, DistributionTypes.error>, 
-      description: string, 
-      floatOp: (float, float) => float, 
-      dist1': result<SymbolicDistTypes.symbolicDist, string>, 
-      dist2': result<SymbolicDistTypes.symbolicDist, string>
-    ) => {
-    let dist1 = dist1'->E.R2.fmap(x=>DistributionTypes.Symbolic(x))->E.R2.fmap2(s=>DistributionTypes.Other(s))
-    let dist2 = dist2'->E.R2.fmap(x=>DistributionTypes.Symbolic(x))->E.R2.fmap2(s=>DistributionTypes.Other(s))
+    distOp: (
+      DistributionTypes.genericDist,
+      DistributionTypes.genericDist,
+    ) => result<DistributionTypes.genericDist, DistributionTypes.error>,
+    description: string,
+    floatOp: (float, float) => float,
+    dist1': SymbolicDistTypes.symbolicDist,
+    dist2': SymbolicDistTypes.symbolicDist,
+    ~epsilon: float,
+  ) => {
+    let dist1 = dist1'->DistributionTypes.Symbolic // ->DistributionTypes.Other
+    let dist2 = dist2'->DistributionTypes.Symbolic // ->DistributionTypes.Other
     let received =
-      E.R.liftJoin2(distOp, dist1, dist2)
-      ->E.R2.fmap(mean)
-      ->E.R2.fmap(run)
-      ->E.R2.fmap(toFloat)
+      distOp(dist1, dist2)->E.R2.fmap(mean)->E.R2.fmap(run)->E.R2.fmap(toFloat)->E.R.toExn
     let expected = floatOp(runMean(dist1), runMean(dist2))
     switch received {
-    | Error(err) => impossiblePath(description)
-    | Ok(x) =>
-      switch x {
-      | None => impossiblePath(description)
-      | Some(x) => x->expect->toBeSoCloseTo(expected, ~digits)
-      }
+    | None => expectImpossiblePath(description)
+    | Some(x) => expectErrorToBeBounded(x, expected, ~epsilon)
     }
   }
+}
 
-  describe("addition", () => {
-    let testAdditionMean = testOperationMean(algebraicAdd, "algebraicAdd", \"+.")  
+let {
+  algebraicAdd,
+  algebraicMultiply,
+  algebraicDivide,
+  algebraicSubtract,
+  algebraicLogarithm,
+  algebraicPower,
+} = module(DistributionOperation.Constructors)
 
-    testAll("homogeneous addition", zipDistsDists, dists => {
-      let (dist1, dist2) = dists
-      testAdditionMean(dist1, dist2)
+let algebraicAdd = algebraicAdd(~env)
+let algebraicMultiply = algebraicMultiply(~env)
+let algebraicDivide = algebraicDivide(~env)
+let algebraicSubtract = algebraicSubtract(~env)
+let algebraicLogarithm = algebraicLogarithm(~env)
+let algebraicPower = algebraicPower(~env)
+
+let {testOperationMean, distributions, pairsOfDifferentDistributions, epsilon} = module(Internals)
+
+describe("Means invariant", () => {
```

Comment:
Maybe "Means are invariant" or something a bit more explanatory?

Reply:
I'm not intending the `describe(` block to be read on it's own, I'm thinking about the sentence made from the nested `describe(`s then the `test(` statements.

Reply:
I was thinking that too, I just think "Means are invariant for addition..." sounds better in a sentence

Reply:
oh! I get it now.

Reply:
next push.